### PR TITLE
feature/sc-121627/make-particle-update-command-uses-the-new

### PR DESCRIPTION
**Pull request by hugomontero**
## Description
Make `particle update` uses the new flash methods

## How to Test
**Try the follow cases by connecting your device in Normal mode and then in DFU mode**
1. `particle update`
    * It will take the latest device OS version for the device and flash it
    * In case you have 2 o more device connected it will prompt to you to pick one
2. `particle update ${deviceID}`
    * It will take the latest device OS version for the specified device and flash it
    * In case you have 2 o more devices it will ignore it and look for the specified one.
3. `particle update --target 3.3.0`
    * It will get all binaries for 3.3.0 device os version and flash it into the device
    * In case you have 2 o more device connected it will prompt to you to pick one
4. `particle update --target 3.3.0 ${deviceID}`
    * It will get all binaries for 3.3.0 device os version and flash it into the device.
    * In case you have 2 o more devices it will ignore it and look for the specified one.

## Completeness

- [x] User is totes amazing for contributing!
- [x] Contributor has signed [CLA](https://docs.google.com/a/particle.io/forms/d/1_2P-vRKGUFg5bmpcKLHO_qNZWGi5HKYnfrrkd-sbZoA/viewform)
- [x] Problem and solution clearly stated
- [x] Tests have been provided
- [x] Docs have been updated
- [x] CI is passing

### CODE REVIEW

File: src/cmd/update.js [hunk 1]
```diff
@@ -1,255 +1,44 @@
...
-				const platformName = isKnownPlatformId(d.platformId) ? platformForId(d.platformId).displayName : `Platform ${d.platformId}`;
-				return {
-					name: `${d.id} (${platformName})`,
-					value: d
-				};
-			})
+	async updateDevice(deviceIdOrName, { target }) {
+		const { api, auth , particleApi } = this._particleApi();
+		// get device info
+		const device = await usbUtils.getOneUsbDevice({ deviceIdOrName, api, auth, ui: this.ui });
```
Comment (keeramis) [inline src/cmd/update.js:19]: Same as this :)
https://github.com/particle-iot/particle-cli/pull/664

File: src/cmd/update.js [hunk 2]
```diff
@@ -1,255 +1,44 @@
...
+		const version = target || 'latest';
+		// get platform info
```
Comment (keeramis) [inline src/cmd/update.js:21]: If user gives the target as a garbage string, I noticed that it interestingly converts it to a device-os versions. Perhaps out of scope of particle-cli and something to look in the API service.
```
 $ npm start -- update --target asdf

> particle-cli@3.12.0 start
> node ./src/index.js update --target asdf

? Which device would you like to select? p2
Updating p2 0a10aced202194944a02c4d4 with version asdf
[████████████████████████████████████████] 100% | Flashing p2-system-part1@5.4.0.bin
```

File: src/cmd/update.js (hunk 1 above)
Comment (hugomontero) [reply src/cmd/update.js]: Fixed here! https://github.com/particle-iot/particle-cli/pull/663/commits/49f313182606532d0000efb451c0c04265963262

File: src/cmd/update.js (hunk 2 above)
Comment (hugomontero) [reply src/cmd/update.js:28]: I've added a simple semver validation over here https://github.com/particle-iot/particle-cli/pull/663/commits/49f313182606532d0000efb451c0c04265963262

**Review by keeramis — APPROVED** [review]
Great work! 🎉

**Review by monkbroc — APPROVED** [review]
Nice work!

**Pull request merged**